backend/app/Observers/UserRelationObserver.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Observers;
+
+use App\Models\UserNotification;
+use App\Models\UserRelation;
+
+class UserRelationObserver
+{
+    public $afterCommit = true;
+
+    public function created(UserRelation $userRelation)
+    {
+        UserNotification::create([
+            'user_id' => $userRelation->following_id,
+            'notifiable_type' => 'App\Models\UserRelation',
+            'notifiable_id' => $userRelation->id,
+        ]);
+    }
+
+    public function deleted(UserRelation $userRelation)
```

Comment:
also I think when the relation is deleting you should not delete the notifcation, i think you should add a new notification saying that the relation was removed

Reply:
Kept this as is for now so that users won't be receiving an `unfollowed you` notification.